src/player.cpp

```diff
@@ -114,6 +114,46 @@ ItemStack &Player::getWieldedItem(ItemStack *selected, ItemStack *hand) const
 	return (hand && selected->name.empty()) ? *hand : *selected;
 }
 
+bool Player::getOffhandWieldedItem(ItemStack *offhand, ItemStack *place, IItemDefManager *idef, const PointedThing &pointed) const
+{
+	assert(offhand);
+
+	ItemStack main;
+
+	const InventoryList *mlist = inventory.getList("main");
+	const InventoryList *olist = inventory.getList("offhand");
+
+	if (olist)
```

Comment:
`&& olist->getSize() > 0` (can a list be empty?)

Reply:
No, setting the size to 0 is equivalent to deleting the list.

https://github.com/minetest/minetest/blob/981d79157a64468d19f140f5c55d8ccd3855fa71/src/script/lua_api/l_inventory.cpp#L126